tracer/src/Datadog.Trace/IAST/IastRequestContext.cs

```diff
@@ -62,6 +62,12 @@ private void AddQueryPath(string path)
         _taintedObjects.TaintInputString(path, new Source(SourceType.GetByte(SourceTypeName.RequestPath), null, path));
     }
 
+    // Used ONLY for testing purposes
+    internal void AddTaintedForTest(string name, string value, SourceTypeName sourceType = SourceTypeName.RequestHeader)
```

Comment:
Seems to only support one test, so not sure I see a huge amount of value adding it our main assembly.

Reply:
This is a mechanism that we will use a lot when testing propagation and vulnerabilities that use propagation. The idea was to provide a mechanism in order to test that things easily in the next PRs. Still, I guess I could try to remove this method and do the same thing by using reflection. I will check that.

Reply:
I have tried and we can use reflection and remove this method. I have uploaded the code. I agree that this is a better solution. Thanks for the suggestion!